### DISCUSSION TITLE
Python only pysolid package

### DISCUSSION THREAD

**Discussion by piyushrpt** [insarlab/PySolid#99] [Ideas]
This is related to https://github.com/isce-framework/isce3/issues/22
@yunjunz @hfattahi 

Currently, `pysolid` is the only dependency for `isce3` that requires us to install a Fortran compiler to build from source. It would be great if this dependence on `Fortran` can be eliminated.

I did a brute force translation to python and fiddled with the code a bit to give it some structure while retaining variable names, and have a reasonably working python implementation. By working, I mean it passes the tests that are currently in this repo. We would want to test this more extensively to convince ourselves that this is a reasonable replacement.  The computations are fairly straightforward and I don't think performance will be an issue.

How would we go about this comparison?

1. Add a folder called `pyimpl` in `pysolid` and for the test scripts just use `from pysolid.pyimpl import point, grid` . This allows for code to be in same repo and for comparisons.

2. Create a new repo called `pysolid2` and scripts can use `import pysolid2 as pysolid` for the tests.

3. Drop the code into `isce3.solid_earth_tides` and scripts can use `from isce3.solid_earth_tides import pysolid`

**Comment by yunjunz** [comment]
That's awesome @piyushrpt!!! Having a pure python version would be so much simpler. 

For an more comprehensive testing, I would think letting the two versions of code generate several global solid Earth tides time series of several years long, crossing a few dacades, in coarse grid, would be ensuring enough to believe the two are behaving consistently.

As for the comparison, either of the 3 sounds good to me. Once all the tests/comparisons passed, we could replace the current version with the new one in this same original repo. I have given your account the manager permission, and please do not hesitate to let me know if more is needed.
